t/perf/p7822-grep-perl-character.sh

```diff
@@ -7,6 +7,8 @@ etc.) we will test the patterns under those numbers of threads.
 "
 
 . ./perf-lib.sh
+#. "$TEST_DIRECTORY"/lib-gettext.sh
+export LC_ALL=is_IS.utf8
```

Comment:
this is not portable shell, but neither it is meant to be the final code.

ideally, there should be a way to define the right LOCALE (just like lib-gettext.sh) does by proving, and this test should be probably "skipped" if there is no unicode locale to use, as the result aren't "relevant" in that case (at least to compare it with the previous version).

truth is that this test is too noisy to be relevant anyway, so I'd been doing instead tests with `hyperfine` that show at least 20% improvement by restricting the sets to ASCII